appium-tests/ios/ios.spec.js

```diff
@@ -85,204 +84,187 @@ describe('Camera tests iOS.', function () {
-        var command = "navigator.camera.getPicture(function (result) { document.getElementById('info').innerHTML = 'Success: ' + result.slice(0, 100); }, " +
-                      "function (err) { document.getElementById('info').innerHTML = 'ERROR: ' + err; }," + JSON.stringify(options) + ");";
+
         return driver
-            .sleep(2000)
             .context(webviewContext)
-            .execute(command)
-            .sleep(5000)
+            .execute(cameraHelper.getPicture, [options, promiseId])
             .context('NATIVE_APP')
             .then(function () {
                 if (skipUiInteractions) {
                     return;
                 }
                 if (options.hasOwnProperty('sourceType') && options.sourceType === cameraConstants.PictureSourceType.PHOTOLIBRARY) {
                     return driver
-                        .elementByName('Camera Roll')
+                        .waitForElementByXPath('//*[@label="Camera Roll"]', MINUTE / 2)
                         .click()
                         .elementByXPath('//UIACollectionCell')
                         .click()
                         .then(function () {
-                            if (options.hasOwnProperty('allowEdit') && options.allowEdit === true) {
+                            if (options.allowEdit) {
                                 return driver
-                                    .elementByName('Use')
-                                    .click();
+                                    .elementByXPath('//*[@label="Use"]')
+                                    .click()
+                                    .fail(function () {
+                                        return driver
+                                            .elementByXPath('//UIAButton[@label="Choose"]')
+                                            .getLocation()
+                                            .then(function (loc) {
+                                                var tapChoose = new wd.TouchAction();
+                                                tapChoose.tap(loc);
+                                                return driver
+                                                    .performTouchAction(tapChoose);
+                                            });
+                                    });
                             }
                             return driver;
                         });
                 }
                 if (options.hasOwnProperty('sourceType') && options.sourceType === cameraConstants.PictureSourceType.SAVEDPHOTOALBUM) {
                     return driver
-                        .elementByXPath('//UIACollectionCell')
+                        .waitForElementByXPath('//UIACollectionCell', MINUTE / 2)
                         .click()
                         .then(function () {
-                            if (options.hasOwnProperty('allowEdit') && options.allowEdit === true) {
+                            if (options.allowEdit) {
```

Comment:
Please invert this statement to reduce nesting. The code inside looks overly indented.